src/lazy/expanded/macro_table.rs

```diff
@@ -159,14 +158,14 @@ impl<'top> MacroRef<'top> {
 /// its validity and allowing evaluation to begin.
 #[derive(Debug, Clone)]
 pub struct MacroTable {
-    macros_by_address: Vec<Macro>,
+    macros_by_address: Vec<Rc<Macro>>,
```

Comment:
🗺️ The `MacroTable` now stores `Rc<Macro>`s, so copying the macro table's contents to a new macro table is substantially cheaper than it was before.